src/docfx/lib/YamlUtility.cs

```diff
@@ -166,35 +166,39 @@ private static JToken ToJson(YamlNode node, JTokenSourceMap mappings = null)
                 {
                     if (key is YamlScalarNode scalarKey)
                     {
-                        var jToken = ToJson(value, mappings);
+                        var path = string.Join('|', name, key.ToString());
+                        SetMappings(mappings, key, path);
```

Comment:
how do we put these line info to `JToken` if it is a separate dictionary?

Reply:
For flat types, we can just use "|propertyName". For nested types during the custom validation we can build the structure separated with "|" as the mapping key